vast/vast.cpp

```diff
@@ -50,9 +80,51 @@ int main(int argc, char** argv) {
     return EXIT_FAILURE;
   }
   // Application setup.
-  const auto [root, factory] = make_application(argv[0]);
+  auto [root, root_factory] = make_application(argv[0]);
   if (!root)
     return EXIT_FAILURE;
+  // Load plugins.
+  auto& plugins = plugins::get();
+  auto plugin_dirs = detail::get_plugin_dirs(cfg);
+  for (const auto& dir : plugin_dirs) {
+    VAST_VERBOSE_ANON("looking for plugins in", dir);
+    for (const auto& file : directory{dir}) {
+      if (file.extension() == ".so" || file.extension() == ".dylib") {
+        if (auto plugin = plugin_ptr{file.str().c_str()}) {
+          VAST_VERBOSE_ANON("loaded plugin:", file);
+          plugins.push_back(std::move(plugin));
+        } else {
+          VAST_ERROR_ANON("failed to load plugin", file);
+        }
+      }
+    }
+  }
+  // Initialize successfully loaded plugins.
+  for (auto& plugin : plugins) {
+    VAST_VERBOSE_ANON("initializing plugin:", plugin->name());
+    auto key = "plugins."s + plugin->name();
```

Comment:
I think we need to handle the case where two plugins have the same name. (e.g. due to copy/pasting the example)

Reply:
Fair point, we'll error in that case and unload the plugin.

Reply:
...and we should bail out as soon as we have the first duplicate plugin name.